### PR TITLE
Added Coordinators for Tab Bar

### PR DESCRIPTION
**Pull request by mlchct**
## Summary 
* Added the foundation of MVVM - C
## Changes
* Created coordinators for app launch, tab bar and tab bar screens
* Created mock pages for other tabs
* Moved app launch logic in AppDelegate

### CODE REVIEW

File: Instagram-like-trainee/Coordinators/AppCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class AppCoordinator: CoordinatorProtocol {
+    
+    var childCoordinators = [CoordinatorProtocol]()
+    var navigationController: UINavigationController
+   
+    var rootViewController: UIViewController {
+        self.navigationController
+    }
+    
+    init(rootViewController: UINavigationController) {
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/AppCoordinator.swift:19]: ViewController is not a NavigationController. Naming mistake?

File: Instagram-like-trainee/Coordinators/AppCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import UIKit
+
+class AppCoordinator: CoordinatorProtocol {
+    
+    var childCoordinators = [CoordinatorProtocol]()
+    var navigationController: UINavigationController
+   
+    var rootViewController: UIViewController {
+        self.navigationController
+    }
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/AppCoordinator.swift:17]: Should it be `private`?

File: Instagram-like-trainee/Coordinators/AppCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
+//
+//  AppCoordinator.swift
+//  Instagram-like-trainee
+//
+//  Created by user on 04/04/2024.
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/AppCoordinator.swift:5]: Created by **user** on 04/04/2024.
Please add your name to identify yourself

File: Instagram-like-trainee/Coordinators/MainCoordinator.swift [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+//  Created by user on 03/04/2024.
+//
+
+import UIKit
+
+class MainCoordinator: CoordinatorProtocol{
+    
+    weak var parentCoordinator: AppCoordinator?
+    var childCoordinators = [CoordinatorProtocol]()
+    var tabBarController: UITabBarController
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/MainCoordinator.swift:14]: `Private`? Please check all this issues

File: Instagram-like-trainee/Coordinators/MainCoordinator.swift [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    
+    private func configureTabBarItem(
+        viewController: UINavigationController,
+        image: UIImage?,
+        selectedImage: UIImage?
+    ) -> UINavigationController{
+        viewController.tabBarItem.image = image
+        viewController.tabBarItem.selectedImage = selectedImage
+        return viewController
+    }
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/MainCoordinator.swift:92]:
```suggestion
    private func configureTabBarItem(
        viewController: UINavigationController,
        image: UIImage?,
        selectedImage: UIImage?
    ) -> UINavigationController{
        viewController.tabBarItem.image = image
        viewController.tabBarItem.selectedImage = selectedImage
        return viewController
    }
```

File: Instagram-like-trainee/Coordinators/TabsCoordinators/AddPostCoordinator.swift [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+//  Instagram-like-trainee
+//
+//  Created by user on 04/04/2024.
+//
+
+import UIKit
+
+protocol AddPostCoordinatorProtocol: CoordinatorProtocol{
+    
+}
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/AddPostCoordinator.swift:12]:
```suggestion
protocol AddPostCoordinatorProtocol: CoordinatorProtocol {}
```

File: Instagram-like-trainee/Coordinators/TabsCoordinators/AddPostCoordinator.swift [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+class AddPostCoordinator: AddPostCoordinatorProtocol {
+    weak var parentCoordinator: MainCoordinator?
+    var childCoordinators = [CoordinatorProtocol]()
+    var navigationController: UINavigationController
+
+    var rootViewController: UIViewController {
+        navigationController
+    }
+
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/AddPostCoordinator.swift:22]: Empty line

File: Instagram-like-trainee/Coordinators/TabsCoordinators/AddPostCoordinator.swift [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    
+    init(rootViewController: UINavigationController) {
+        self.navigationController = rootViewController
+    }
+    
+    func start() {
+        
+    }
+    
+
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/AddPostCoordinator.swift:32]: Empty lines

File: Instagram-like-trainee/Coordinators/TabsCoordinators/MyProfileCoordinator.swift [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+//  Instagram-like-trainee
+//
+//  Created by user on 04/04/2024.
+//
+
+import UIKit
+
+protocol MyProfileCoordinatorProtocol: CoordinatorProtocol {
+    
+}
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/MyProfileCoordinator.swift:12]: Fix

File: Instagram-like-trainee/Coordinators/TabsCoordinators/MyProfileCoordinator.swift [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+class MyProfileCoordinator: MyProfileCoordinatorProtocol{
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/MyProfileCoordinator.swift:14]:
```suggestion
class MyProfileCoordinator: MyProfileCoordinatorProtocol {
```

File: Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+//  Instagram-like-trainee
+//
+//  Created by user on 04/04/2024.
+//
+
+import UIKit
+
+protocol ReelsCoordinatorProtocol: CoordinatorProtocol {
+    
+}
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift:12]: Fix

File: Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    
+    weak var parentCoordinator: MainCoordinator?
+    var childCoordinators = [CoordinatorProtocol]()
+    var navigationController: UINavigationController
+
+    var rootViewController: UIViewController {
+        navigationController
+    }
+    
+
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift:24]: Fix

File: Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    
+    init(rootViewController: UINavigationController) {
+        self.navigationController = rootViewController
+    }
+    
+    func start() {
+        
+    }
+    
+
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift:33]: Fix

File: Instagram-like-trainee/Screens/HomePage/View/HomePageViewController.swift [hunk 14]
```diff
@@ -7,7 +7,9 @@
 
 import UIKit
 
-class MainPageViewController: UIViewController {
+class HomePageViewController: UIViewController,HomeScreenViewProtocol {
+    
+    var coordinator: HomeCoordinatorProtocol?
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Screens/HomePage/View/HomePageViewController.swift:12]:
```suggestion
class HomePageViewController: UIViewController, HomeScreenViewProtocol {
    
    var coordinator: HomeCoordinatorProtocol?
```

File: Instagram-like-trainee/Screens/TabBarScreenViewProtocol.swift [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+
+import Foundation
+
+protocol TabBarScreenViewProtocol{
+    
+}
+
+protocol TabBarScreenViewModelDelegateProtocol{
+    
+}
```
Comment (Danila-Bernat-iOS) [inline Instagram-like-trainee/Screens/TabBarScreenViewProtocol.swift:16]: Fix

File: Instagram-like-trainee/Coordinators/AppCoordinator.swift (hunk 1 above)
Comment (mlchct) [reply Instagram-like-trainee/Coordinators/AppCoordinator.swift:19]: Done

File: Instagram-like-trainee/Coordinators/MainCoordinator.swift (hunk 4 above)
Comment (mlchct) [reply Instagram-like-trainee/Coordinators/MainCoordinator.swift:14]: Done

File: Instagram-like-trainee/Coordinators/AppCoordinator.swift (hunk 3 above)
Comment (mlchct) [reply Instagram-like-trainee/Coordinators/AppCoordinator.swift:5]: Done

File: Instagram-like-trainee/Screens/TabBarScreenViewProtocol.swift (hunk 15 above)
Comment (mlchct) [reply Instagram-like-trainee/Screens/TabBarScreenViewProtocol.swift:16]: Done

File: Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift (hunk 13 above)
Comment (mlchct) [reply Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift:33]: Done

File: Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift (hunk 12 above)
Comment (mlchct) [reply Instagram-like-trainee/Coordinators/TabsCoordinators/ReelsCoordinator.swift:24]: Done

File: Instagram-like-trainee/Screens/HomePage/View/HomePageViewController.swift (hunk 14 above)
Comment (mlchct) [reply Instagram-like-trainee/Screens/HomePage/View/HomePageViewController.swift:12]: Done

File: Instagram-like-trainee/Coordinators/MainCoordinator.swift (hunk 5 above)
Comment (mlchct) [reply Instagram-like-trainee/Coordinators/MainCoordinator.swift:92]: Done

**Review by Danila-Bernat-iOS — APPROVED** [review]

**Review by e-garbuzov-wise — APPROVED** [review]

**Pull request merged**